app/models/manageiq/providers/google/cloud_manager/refresh_parser.rb

```diff
@@ -61,8 +62,31 @@ def get_images
         process_collection(images, :vms) { |image| parse_image(image) }
       end
 
+      def get_key_pairs(instances)
+        ssh_keys = []
+
+        # Find all key pairs added directly to GCE instances
+        instances.each do |instance|
+          ssh_keys |= parse_compute_metadata_ssh_keys(instance.metadata)
+        end
+
+        # Add ssh keys that are common to all instances in the project
+        project_common_metadata = @connection.projects.get(@ems.project).common_instance_metadata
```

Comment:
Is this something that could be retrieved and cached?  I worry about making this call for each instance.